instrumentation/micrometer/micrometer-1.5/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/UnsupportedReadLogger.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.micrometer.v1_5;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class UnsupportedReadLogger {
+
+  private static final Logger logger = LoggerFactory.getLogger(OpenTelemetryMeterRegistry.class);
+  private static final AtomicBoolean done = new AtomicBoolean(false);
+
+  static void logWarning() {
+    if (done.compareAndSet(false, true)) {
```

Comment:
Instead you can put the log message in a static initializer, which makes sure it's only logged once on classload, which happens lazily while calling from the Unsupported method

Reply:
Yeah, that's a really good idea, will do that. Thanks!